instrumentation/CONTRIBUTING.md

```diff
@@ -151,11 +194,34 @@ When you do in fact run into cases where test doubles or API stubs are absolutel
 
 ### Understand Performance Characteristics
 
-Instrumentation libraries should be as lightweight as possible
+Instrumentation libraries should be as lightweight as possible and must:
 
-* Avoid allocating too many objects and private methods
+* Avoid allocating objects unless absolutely necessary
```

Comment:
Maybe it's just my personal opinion, but when I read "unless absolutely necessary", I will compromise everything else (including code legibility and possibly maintainability) in order to achieve the reduction in object allocation.

I have *a* suggestion, which may not be perfect, but it kind of captures how I think it would make sense to convey this advice (but please feel free to change it):
```suggestion
* Only allocate objects are necessary
```